src/base/Context.ts

```diff
@@ -226,6 +228,8 @@ export class Context {
 
     try {
       newCurrentPair = keyring.getPair(address);
+
+      isPrefixValid(address, this.ss58Format);
```

Comment:
Better put this before the assignment (I know it's mostly the same functionally, it's just a matter of semantics, the idea being that you don't assign the value if it's an incorrect one)